dataprofiler/profilers/profile_builder.py

```diff
@@ -444,6 +445,361 @@ def clean_data_and_get_base_stats(df_series, sample_size,
+
+    def _update_profile_from_chunk(self, data, sample_size,
+                                   min_true_samples=None):
+        """
+        Iterate over the columns of a dataset and identify its parameters.
+
+        :param data: a text dataset
+        :type data: Union[pd.Series, pd.DataFrame, list]
+        :param sample_size: number of samples for df to use for profiling
+        :type sample_size: int
+        :param min_true_samples: minimum number of true samples required
+        :type min_true_samples: int
+        :return: list of column profile base subclasses
+        :rtype: list(BaseColumnProfiler)
+        """
+
+        if isinstance(data, pd.DataFrame):
+            if len(data.columns) > 1:
+                raise ValueError("The unstructured cannot handle a dataset "
+                                 "with more than 1 column. Please make sure "
+                                 "the data format of the dataset is "
+                                 "appropriate.")
+            data = data[data.columns[0]]
+        elif isinstance(data, list):
+            # we know that if it comes in as a list, it is a 1-d list based
+            # bc of our data readers
+            data = pd.Series(data)
+
+        # Format the data
+        notification_str = "Finding the empty lines in the data..."
+        print(notification_str)
+        data, base_stats = self._clean_data_and_get_base_stats(
+            data, sample_size, min_true_samples)
+        self._update_base_stats(base_stats)
+
+        if sample_size < len(data):
+            warnings.warn("The data will be profiled with a sample size of {}. "
+                          "All statistics will be based on this subsample and "
+                          "not the whole dataset.".format(sample_size))
+
+        # process the text data
+        notification_str = "Calculating the statistics... "
+        print(notification_str)
+        pool = None
+        if self._profile is None:
+            # TODO: update when unstructured_options becomes available
+            self._profile = UnstructuredCompiler(
+                data, options=self.options.structured_options, pool=pool)
+        else:
+            self._profile.update_profile(data, pool=pool)
+
+    def update_profile(self, data, sample_size=None, min_true_samples=None):
+        """
+        Update the profile for data provided. User can specify the sample
+        size to profile the data with. Additionally, the user can specify the
+        minimum number of non-null samples to profile.
+
+        :param data: data to be profiled
+        :type data: Union[data_readers.base_data.BaseData, pandas.DataFrame]
+        :param sample_size: number of samples to profile from the data
+        :type sample_size: int
+        :param min_true_samples: minimum number of non-null samples to profile
+        :type min_true_samples
+        :return: None
+        """
+        if isinstance(data, data_readers.base_data.BaseData):
```

Comment:
Can't this only handle `TextData` anyway, so can we make this `isinstance` check more rigorous?

Reply:
We could limit it to be that, but technically it could be CSVData depending on the data format

Reply:
i.e. `records` would work